### PR TITLE
wip: add search-in-workspace UI tests

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I had to update `chromedriver` in order for certain api to work as intended (ex: input.addValue()), but it seems that travis and selenium must also be updated in order for the ci to work. @simark do you have any thoughts?

**Comment by akosyakov** [comment]
@vince-fugnitto just fyi: please have a look at [PageObject](https://martinfowler.com/bliki/PageObject.html) pattern - an idea that tests should not be written against dom/css but app api which is implemented via dom/css, it allows to encapsulate flaky code (css selectors) in one place and reuse app api across tests

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto just fyi: please have a look at [PageObject](https://martinfowler.com/bliki/PageObject.html) pattern - an idea that tests should not be written against dom/css but app api which is implemented via dom/css, it allows to encapsulate flaky code (css selectors) in one place and reuse app api across tests

Thanks @akosyakov, I'll take a look and make the necessary changes :)

**Comment by vince-fugnitto** [comment]
@akosyakov I've made some updates, was this more what you had in mind regarding the **PageObject** pattern?

File: examples/browser/test/workspace/search-in-workspace.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+export class SearchInWorkspace {
+
+    /** accessors */
+    private readonly _searchField = '#search-input-field';
+    private readonly _includeField = '#include-glob-field';
+    private readonly _excludeField = '#exclude-glob-field';
+
+    private readonly _searchInfo = '.search-info';
+    private readonly _options = '//*[@id="search-in-workspace"]/div/div[1]/div[1]/div[3]/div[1]/span';
```
Comment (akosyakov) [inline examples/browser/test/workspace/search-in-workspace.ts]: Can we add id to span to look up it easier and make tests less dependent on dom?

**Comment by akosyakov** [comment]
@vince-fugnitto looks good, yes, I meant something like that :)

it is fine with me if @elaihau finishes the review

File: examples/browser/test/workspace/search-in-workspace.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { TopPanel } from '../top-panel/top-panel';
+import { LeftPanel } from '../left-panel/left-panel';
+
+export class SearchInWorkspace {
+
+    /** accessors */
```
Comment (elaihau) [inline examples/browser/test/workspace/search-in-workspace.ts]: the following is the definition of "accessor" from google:

```
assessor
/əˈsesər/
noun
noun: assessor; plural noun: assessors
a person who evaluates the quality of a person or thing.
"the assessor interviews the candidates and observes them actually using the language"
a person who calculates or estimates the price or value of something.
"loss assessors were beginning to get an idea of the extent of the damage"
a person who is knowledgeable in a particular field and is called upon for advice, typically by a judge or committee of inquiry.
```
i guess you meant "selectors" ?

Comment (vince-fugnitto) [reply examples/browser/test/workspace/search-in-workspace.ts]: sorry yes, thank you!

**Comment by marcdumais-work** [comment]
@vince-fugnitto Can you look into [this](https://ci.appveyor.com/project/kittaakos/theia/builds/21452128#L2789) AppVeyor error? Looks related to your new tests.

**Comment by marcdumais-work** [comment]
Also, the Travis tests ended-up passing on the second and last re-try (3rd try).

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto Can you look into [this](https://ci.appveyor.com/project/kittaakos/theia/builds/21452128#L2789) AppVeyor error? Looks related to your new tests.

Yes, thank you @marcdumais-work  I will take a look. I'm keeping the PR under WIP until I can address all the flakiness in the CI, I don't want to contribute more problems in our builds :(

**Comment by vince-fugnitto** [comment]
closing for the moment to wait and see for updating to cypress ui tests.

**Pull request closed without merging**